packages/jest-mock-apollo/src/GraphQLClientFactory.ts

```diff
@@ -0,0 +1,95 @@
+import {buildSchema, GraphQLSchema} from 'graphql';
+import {GraphQLRequest, ApolloLink} from 'apollo-link';
+import {
+  ApolloReducerConfig,
+  InMemoryCache,
+  IntrospectionFragmentMatcher,
+} from 'apollo-cache-inmemory';
+import ApolloClient from 'apollo-client';
+import {GraphQLMock, GraphQLClientConfig, GraphQLClientOptions} from './types';
+import MockApolloLink from './MockApolloLink';
+import Requests from './Requests';
+
+export type MockGraphQLClient = ApolloClient<any> & {
+  graphQLRequests: Requests;
+  graphQLResults: Promise<any>[];
+};
+
+function defaultGraphQLMock({operationName}: GraphQLRequest) {
+  return new Error(
+    `Can’t perform GraphQL operation '${operationName ||
+      ''}' because no mocks were set.`,
+  );
+}
+
+export default class GraphQLClientFactory {
+  private unionOrIntersectionTypes: any[];
+  private schema: GraphQLSchema;
+  private cacheOptions: ApolloReducerConfig;
+
+  constructor({
+    unionOrIntersectionTypes = [],
+    schema,
+    schemaSrc,
```

Comment:
Why take both?

Reply:
I thought that we want to allow for passing in an arbitrary schema? Are you saying we should make every consumer build their schema and pass it in?